factory/statusCore/statusCoreComponents.go

```diff
@@ -67,19 +67,10 @@ func NewStatusCoreComponentsFactory(args StatusCoreComponentsFactoryArgs) (*stat
 
 func checkArgs(args StatusCoreComponentsFactoryArgs) error {
 	if check.IfNil(args.CoreComp) {
-		return errErd.ErrNilCoreComponents
+		return errorsMx.ErrNilCoreComponents
 	}
 	if check.IfNil(args.CoreComp.EconomicsData()) {
-		return errErd.ErrNilEconomicsData
-	}
-	if check.IfNil(args.CoreComp.GenesisNodesSetup()) {
```

Comment:
Not needed to check for nil GenesisNodesSetup, InternalMarshalizer and Uint64ByteSliceConverter ?

Reply:
same reason here.. they are only forwarded to subcomponents